v2/pkg/founat/client.go

```diff
@@ -309,24 +323,73 @@ func (c *natClient) AddEgress(link netlink.Link, subnets []*net.IPNet) error {
 		}
 	}
 
+	for _, r := range deletes {
+		if c.logFunc != nil {
+			c.logFunc(fmt.Sprintf("removing a destination %s", r.Dst.String()))
+		}
+		if err := netlink.RouteDel(&r); err != nil {
+			return fmt.Errorf("netlink: failed to delete a route  %+v, %w", r, err)
+		}
+	}
+
 	for _, n := range adds {
 		if err := c.addEgress1(link, n); err != nil {
 			return err
 		}
+
+		family := iptables.ProtocolIPv6
+		if n.IP.To4() != nil {
+			family = iptables.ProtocolIPv4
+		}
+
+		if originatingOnly {
+			if err := configureRoutes(family, c.backend); err != nil {
+				return err
+			}
+		}
 	}
 
-	for _, r := range deletes {
-		if c.logFunc != nil {
-			c.logFunc(fmt.Sprintf("removing a destination %s", r.Dst.String()))
+	if originatingOnly {
+		for _, f := range getFamilies(subnets) {
+			if err := configureRoutes(f, c.backend); err != nil {
+				return err
+			}
 		}
```

Comment:
@p-strusiewiczsurmacki-mobica I have a question about `configureRoutes()` - Why is it being called inside the loop over the adds variable? 
My understanding is that calling it once should be sufficient, so I wonder why is it being called in a loop rather than just once at L354.